concreate/generator.py

```diff
@@ -19,9 +19,16 @@
 class Generator(object):
 
     def __init__(self, descriptor_path, target, overrides):
-
-        self.descriptor = Image(tools.load_descriptor(descriptor_path),
-                                os.path.dirname(descriptor_path))
+        descriptor = tools.load_descriptor(descriptor_path)
+
+        # if there is a local modules directory and no modules are defined
+        # we will inject it for a backward compatibility
+        local_mod_path = os.path.join(os.path.abspath(os.path.dirname(descriptor_path)), 'modules')
+        if os.path.exists(local_mod_path) and 'modules' in descriptor:
+            modules = descriptor.get('modules')
+            if not modules.get('repositories'):
```

Comment:
Will it not throw KeyError? Should be: `if not modules.get('repositories', null):`

Reply:
Or `{}`

Reply:
nope, None is default IIRC

Reply:
Nope: https://docs.python.org/2/library/stdtypes.html#typesmapping

get(key[, default])
Return the value for key if key is in the dictionary, else default. If default is not given, it defaults to None, so that this method never raises a KeyError.

Reply:
You are right in fact. I read the docs too fast, sorry!